packages/admin/src/Resources/Pages/ListRecords/Concerns/CanCreateRecords.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Filament\Resources\Pages\ListRecords\Concerns;
+
+use Filament\Pages\Actions\ButtonAction;
+use Filament\Pages\Actions\Modal;
+use Filament\Resources\Pages\Concerns\UsesResourceForm;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
+
+trait CanCreateRecords
+{
+    use UsesResourceForm;
+
+    protected function hasCreateAction(): bool
+    {
+        if (! static::getResource()::canCreate()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected function getCreateAction(): ButtonAction
+    {
+        return parent::getCreateAction()
+            ->url(null)
+            ->form($this->getCreateFormSchema())
+            ->mountUsing(fn () => $this->fillCreateForm())
+            ->modalActions([
+                Modal\Actions\ButtonAction::make('submit')
+                    ->label(__('filament::resources/pages/list-records.actions.create.modal.actions.create.label'))
+                    ->submit('callMountedAction')
+                    ->color('primary'),
+                Modal\Actions\ButtonAction::make('submit')
```

Comment:
Does it matter that these buttons have the same ID/name?

Reply:
eh, probably not in this case. but it's better practice to have them unique all the time. I will do that now